server/index.js

```diff
@@ -72,6 +72,106 @@ app.get('/api/products/:productId', (req, res, next) => {
     });
 });
 
+// below is for /api/cart
+
+app.get('/api/cart', (req, res, next) => {
+  // gets all productsin cart
+  if (!req.session.cartId) {
+    return res.json([]);
+  }
+  const sql = `
+    select "c"."cartItemId",
+       "c"."price",
+       "p"."productId",
+       "p"."image",
+       "p"."name",
+       "p"."shortDescription"
+      from "cartItems" as "c"
+      join "products" as "p" using ("productId")
+      where "c"."cartId" = $1
+  `;
+  db.query(sql, [req.session.cartId])
+    .then(result => {
+      const products = result.rows;
+      res.json(products);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.post('/api/cart/:productId', (req, res, next) => {
+  // gets all products, but no long description
+  const productId = parseInt(req.params.productId, 10);
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return res.status(400).json({
+      error: '"productId" must be a positive integer'
+    });
+  }
+
+  const sql = `
+    select "p"."price"
+    from "products" as "p"
+    where "p"."productId" = $1
+  `;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows[0].price) { // if there is no price
+        return (new ClientError(`cannot find price of productId ${productId}`), 404);
+      }
```

Comment:
Instead of `return`, you'll need to `throw` in order to shortcut this promise chain to its final `.catch()` and prevent the rest of the `.then()` callbacks from executing.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw